plugins/modules/azure_rm_registrationassignment.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2020 Fred-Sun, (@Fred-Sun)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_registrationassignment
+version_added: '1.3.0'
+short_description: Manage Azure RegistrationAssignment instance
+description:
+    - Create and delete instance of Azure RegistrationAssignment.
+options:
+    scope:
+        description:
+            - Scope of the registration assignment..
+        required: true
+        type: str
+    registration_assignment_id:
+        description:
+            - ID of the registration assignment.
+        type: str
+    properties:
+        description:
+            - Properties of a registration assignment.
+        type: dict
+        suboptions:
+            registration_definition_id:
+                description:
+                  - Fully qualified path of the registration definition.
+                required: true
+                type: str
+    state:
+        description:
+            - Assert the state of the RegistrationAssignment.
+            - Use C(present) to create or update an RegistrationAssignment and C(absent) to delete it.
+        default: present
+        type: str
+        choices:
+            - absent
+            - present
+extends_documentation_fragment:
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+author:
+    - Fred-Sun (@Fred-Sun)
+
+'''
+
+EXAMPLES = '''
+    - name: Delete Registration Assignment
+      azure_rm_registrationassignment:
+        scope: subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        registration_assignment_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        state: absent
+
+
+    - name: Create Registration Assignment in subscription level
+      azure_rm_registrationassignment:
+        scope: subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        registration_assignment_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        properties:
+          registration_definition_id: /subscriptions/xxx-xxx/providers/Microsoft.ManagedServices/registrationDefinitions/xxx-xxx
+
+
+    - name: Create Registration Assignment in resourcegroup level
+      azure_rm_registrationassignment:
+        scope: subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup
+        registration_assignment_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```

Comment:
remove this line, to indicate assignment id is not mandatory.

Reply:
fixes